CONTRIBUTING.md

```diff
@@ -129,6 +129,42 @@ Run `rush build -t <packagename>` to build a single project, and all local proje
 
 By default, Rush only displays things written to `STDERR`. If you want to see the full output, pass `--verbose` to any of the build commands. This is also true for script commands (see below).
 
+### Testing
+
+Once the project has successfully built, you can test any of our packages by executing `rushx test` from the root of the package that you want to test. This will automatically run the tests in both NodeJS and in a browser. To target these environments individually, you can run `rushx test:node` and `rushx test:browser`.
+
+By default, these npm scripts run recorded tests. We use a custom recording library called `test-utils-recorder`. You can learn more about our recorder at the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).
+
+#### Recorded tests
+
+By default, the npm scripts `test`, `test:node` and `test:browser` run recorded tests in playback mode i.e there is no request made to the service. For http requests made in each test case, there is a recorded response that is used which reproduces the service behavior.
+
+Keep in mind that if you modify any code that is being tested, or any test code, that interacts with the number of network calls that will be made during testing, or the shape of those network calls, the recordings will need to be re-generated.
+
+You can read more in the section: [Regenerating recordings](#regenerating-recordings).
+
+#### Live tests
+
+By default, the npm scripts `test`, `test:node` or `test:browser` run recorded tests. To run live tests, you need the right Azure resources, set appropriate environment variables to point to these resources and set the environment variable `TEST_MODE` to "live". The readme file in the test folder of each package lists the Azure resources you would need and the environment variables to update.
+
+You can create the Azure resources manually on your own or automate this process by...
+
+Our integration tests will run against the live resources, which are determined by the environment variables you provide.
+
+To run the integration tests, you will need to execute `rushx integration-test`.
```

Comment:
Why are we still referring to "integration tests" and the `rushx integration-test` command?
As mentioned elsewhere, we don't need to talk about "unit" vs "integration" tests as we have chosen to talk about "recorded" vs "live" tests. We also don't need to talk about the npm scripts `unit-test` and `integration-test` because a user is expected to only use the `test` npm script. `unit-test` and `integration-test`  are for CI

Reply:
Gotcha, should we change the guidelines (I mean the ones I'm proposing) to rename the unit-test and integration-test commands into: record, playback and live commands? Seems like it would be more straightforward.